src/optim/base.py

```diff
@@ -40,7 +40,8 @@ def train_base(model, opt, data, scheduler, iterations, acc_steps, batch_size, s
         if extra_args.grad_clip != 0.0:
             torch.nn.utils.clip_grad_norm_(model.parameters(), extra_args.grad_clip)
         opt.step()
-        scheduler.step()
+        if scheduler != None:
```

Comment:
nit: if scheduler is not None

(otherwise LGTM)

Reply:
hello, this kind of fix was already proposed in the [soap](https://github.com/epfml/llm-baselines/tree/soap) branch,
but we didn't change the ```sparse.py```

 thanks for the reminder)